### DISCUSSION TITLE
Global before each - how to call cy.visit on every test?

### DISCUSSION THREAD

**Discussion by ghost** [cypress-io/cypress#23755] [General]
So I've just migrated an app to Cypress 10 and with the changes to support files our current solution doesn't work.
What I want to do is have a global beforeEach that gets called on every test by default that visits the main baseUrl. We have defined an index page to test all our parts of the application by simply pressing a button on that page so every test just defines what button to load.

Of course I can add `cy.visit('/')` to every test or have some global function I run, but before 10.0 we could just do these beforeEach commands on the `cypress/plugins/index.ts` and be done with it. But now that doesn't seem to trigger and for the life of me I can't figure out how to run something before every test. We also have some `afterEach` and `after` functionality defined so its not just a single function.

I've looked at the Cypress.Commands.add and the Cypress.on functions but that doesn't seem to do what I need (the .on only has a trigger before the test but thats too soon to start browsing so my page stays empty.

I'm not entirely sure why it seems to be so hard to find how to properly do beforeEach on every test, so I hope with my question it also leads to documentation update so other people can also implement it. I can't find it on any of the examples or tutorials as well. They all seem to trigger cy.visit in their tests and copy paste that every time, which seems like a waste of time when you have hundreds of files.

**Comment by AlanGreene** [comment]
@MartinAtSogeti The `cypress/plugins/index.ts` file is no longer automatically loaded in Cypress 10: https://docs.cypress.io/guides/references/migration-guide#Plugins-File-Removed

According to the docs you can add a global `beforeEach` in any of the support files: https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests#Support-file

I just tested this out on my project and it works as expected.
